compiler/rustc_type_ir/src/outlives.rs

```diff
@@ -230,8 +230,7 @@ pub fn compute_alias_components_recursive<I: Interner>(
         unreachable!("can only call `compute_alias_components_recursive` on an alias type")
     };
 
-    let opt_variances =
-        if kind == ty::Opaque { Some(cx.variances_of(alias_ty.def_id)) } else { None };
+    let opt_variances = cx.opt_alias_variances(kind, alias_ty.def_id);
```

Comment:
existing: the function comment is outdated, taling about a non-existing `parent` arg

also, looking at the call sites, this function should just take an `AliasTy` argument instead of a `Ty` :thinking: